crates/hyperswitch_domain_models/src/type_encryption.rs

```diff
@@ -562,14 +565,242 @@ mod encrypt {
         }
     }
 
+    impl<T> EncryptedJsonType<T>
+    where
+        T: std::fmt::Debug + Clone + serde::Serialize + serde::de::DeserializeOwned,
```

Comment:
What are case T required Debug? is that necessary.

Reply:
The usages of the `Encode` trait methods (in this case `encode_to_vec()`) needs us to implement `Debug`.
Moreover, if the `PaymentMethod` domain model derives `Debug` and one of it's fields has a `EncryptedJsonType<T>`, we'd also need `Debug` to be implemented on `EncryptedJsonType` and thus `T`.